### PR TITLE
Added Cohen Sutherland Line Clipping Algorithm

### LINKED ISSUE #858: Cohen Sutherland Line Clipping Algorithm
I would like to add the Cohen Sutherland algorithm in C/C++

### PR DESCRIPTION
**Pull request by marwahmanbir**
Fixes #858  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Cohen Sutherland Line Clipping Algorithm

#### Languages Used:
- C
- C++

#### Files Added:
- Cohen_Sutherland.c
- Cohen_Sutherland.cpp

### CODE REVIEW

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+#include<stdlib.h>
+#include<math.h>
+#include<graphics.h>
+#include<dos.h>
+ 
+typedef struct coordinate
+{
+    int x,y;
+    char code[4];
+}PT;
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Use Point instead of `PT`.

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
+#include<stdio.h>
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Add more comments.

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    
+    delay(500);
+    p1=setcode(p1);
+    p2=setcode(p2);
+    v=visibility(p1,p2);
+    delay(500);
+    
+    switch(v)
+    {
+    case 0: drawwindow();
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Indentation fix.

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+// Enter x1 and y1
+// 100
+// 100
+
+// Enter x2 and y2
+// 200
+// 200
+
+// Output before line clipping
+
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: There is no output.

Comment (marwahmanbir) [reply Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Actually for the output I'll have to add a picture for before and after clipping of the line. Should I add it in the code? Or should I just mention the link for it?

Comment (jainaman224) [reply Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Add a link to the picture.

**Comment by marwahmanbir** [comment]
@jainaman224 Updated and also added Cohen_Sutherland.cpp

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+// Enter x2 and y2
+// 200
+// 200
+
+// Output before line clipping:
+// https://drive.google.com/file/d/1im2UXnXnRY40ynvx0S-wJZe1S3VkmujL/view?usp=sharing
+
+// Output after line clipping:
+// https://drive.google.com/file/d/1IcQmUDupNMYwKcInY33IefuGn0Snei63/view?usp=sharing
+
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp]: Add only one blank line at the end of each file.

Comment (marwahmanbir) [reply Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp]: I've added a single line only, at the end of both files

Comment (jainaman224) [reply Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp]: These are two lines. See them in your editor.

**Comment by marwahmanbir** [comment]
Updated

**Comment by jainaman224** [comment]
You have added it again in second commit.

**Comment by marwahmanbir** [comment]
Oh.. removed that one now.

**Comment by jainaman224** [comment]
Please have a look at https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html.

**Comment by marwahmanbir** [comment]
On it.

**Comment by marwahmanbir** [comment]
Done :+1:

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+void drawwindow();
+//A function to draw the line to be clipped
+void drawline(Point p1, Point p2);
+Point setcode(Point p);
+int visibility(Point p1, Point p2);
+Point resetendpt(Point p1, Point p2);
+ 
+void main()
+{
+    int gd=DETECT,v,gm;
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Add space around operator (i.e. `=`, `,`).

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+    line(450,350,150,350);
+    line(150,350,150,100);
+}
+ 
+void drawline(Point p1, Point p2)
+{
+    line(p1.x,p1.y,p2.x,p2.y);
+}
+
+//Setting the 4 bit code
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Space after `//`.

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+Point setcode(Point p)
+{
+    Point ptemp;    
+
+    if(p.y<100)
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Space around `<`.

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.c [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+    
+    return(ptemp);
+}
+
+//Checking for visibility of the line
+int visibility(Point p1, Point p2)
+{
+    int i,flag=0;   
+
+    for(i=0;i<4;i++)
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.c]: Space after `for` and `if`.

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+#include<math.h>
+#include<graphics.h>
+#include<dos.h>
+
+//Creating a structure for coordinates
+typedef struct coordinate
+{
+    int x,y;
+    char code[4];
+}Point;
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp]: Space after `}`.

File: Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp [hunk 11]
```diff
@@ -0,0 +1,205 @@
+// Cohen-Sutherland Line Clipping Algorithm in C
```
Comment (jainaman224) [inline Cohen_Sutherland_Algorithm/Cohen_Sutherland.cpp]: Use class to solve problem.

**Pull request closed without merging**